batchflow/models/torch/layers/combine.py

```diff
@@ -103,20 +103,24 @@ def extra_repr(self):
                 res += f',\nafter_ids={self.after_ids}'
         return res
 
-
     def spatial_resize(self, inputs):
         """ Force the same shapes of the inputs, if needed. """
         shape_ = get_shape(inputs[0])
         dim_ = get_num_dims(inputs[0])
-        spatial_shape_ = shape_[-dim_:]
+        target_shape = shape_[-dim_:]
 
         resized = [inputs[0]]
         for item in inputs[1:]:
             shape = get_shape(item)
             dim = get_num_dims(item)
             spatial_shape = shape[-dim:]
-            if dim > 0 and spatial_shape != tuple([1]*dim) and spatial_shape != spatial_shape_:
-                item = center_crop(item, get_shape(inputs[0])[2:])
+            if dim > 0 and spatial_shape != tuple([1] * dim) and spatial_shape != target_shape:
+                if any(np.array(spatial_shape) < np.array(target_shape)):
+                    item = pad(item, spatial_shape, target_shape)
+                    if get_shape(item)[-dim:] == target_shape:
+                        resized.append(item)
+                        continue
+                item = center_crop(item, target_shape, dim)
```

Comment:
What if we exploit the feature of `F.pad` to "unpad" given negative widths.
Then the single call encapuslates both `pad` and `center_crop`

```suggestion
                div, mod = np.divmod(np.array(target_shape) - spatial_shape, 2)
                ix = sum(zip(div, div+mod), ())
                item = F.pad(arr, ix[::-1])
```